prow/config/config_test.go

```diff
@@ -1712,96 +1712,142 @@ postsubmits:
 				return nil
 			},
 		},
+		{
+			name: "InRepoConfigAllowedClusters gets defaulted if unset",
+			verify: func(c *Config) error {
+				if len(c.InRepoConfig.AllowedClusters) != 1 ||
+					len(c.InRepoConfig.AllowedClusters["*"]) != 1 ||
+					c.InRepoConfig.AllowedClusters["*"][0] != kube.DefaultClusterAlias {
+					return fmt.Errorf("expected c.InRepoConfig.AllowedClusters to contain exactly one global entry to allow the buildcluster, was %v", c.InRepoConfig.AllowedClusters)
+				}
+				return nil
+			},
+		},
+		{
+			name: "InRepoConfigAllowedClusters gets defaulted if no global setting",
+			prowConfig: `
+in_repo_config:
+  allowed_clusters:
+    foo/bar: ["my-cluster"]
+`,
+			verify: func(c *Config) error {
+				if len(c.InRepoConfig.AllowedClusters) != 2 ||
+					len(c.InRepoConfig.AllowedClusters["*"]) != 1 ||
+					c.InRepoConfig.AllowedClusters["*"][0] != kube.DefaultClusterAlias {
+					return fmt.Errorf("expected c.InRepoConfig.AllowedClusters to contain exactly one global entry to allow the buildcluster, was %v", c.InRepoConfig.AllowedClusters)
+				}
+				return nil
+			},
+		},
+		{
+			name: "InRepoConfigAllowedClusters doesn't get overwritten",
+			prowConfig: `
+in_repo_config:
+  allowed_clusters:
+    foo/bar: ["my-cluster"]
+`,
+			verify: func(c *Config) error {
+				if len(c.InRepoConfig.AllowedClusters) != 2 ||
+					len(c.InRepoConfig.AllowedClusters["foo/bar"]) != 1 ||
+					c.InRepoConfig.AllowedClusters["foo/bar"][0] != "my-cluster" {
+					return fmt.Errorf("expected c.InRepoConfig.AllowedClusters to contain exactly one entry for foo/bar, was %v", c.InRepoConfig.AllowedClusters)
+				}
+				return nil
+			},
+		},
 	}
 
 	for _, tc := range testCases {
-		// save the config
```

Comment:
Sorry for the unreadable diff here, I just moved everything into a `t.Run` block, its in a dedicated commit so if you just look at the first commit, you wont see it.